packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -157,49 +157,84 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
 
   public dispose(): void {
     this.callClient.offStateChange(this.onClientStateChange);
-    this.callAgent.dispose();
+    this.callAgent.dispose().catch((e) => {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
+    });
   }
 
   public isTeamsCall(): boolean {
     return 'meetingLink' in this.locator;
   }
 
   public queryCameras(): Promise<VideoDeviceInfo[]> {
-    return this.deviceManager.getCameras();
+    try {
+      return this.deviceManager.getCameras();
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
+      return Promise.resolve([] as VideoDeviceInfo[]);
+    }
   }
 
   public queryMicrophones(): Promise<AudioDeviceInfo[]> {
-    return this.deviceManager.getMicrophones();
+    try {
+      return this.deviceManager.getMicrophones();
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
+      return Promise.resolve([] as AudioDeviceInfo[]);
+    }
   }
 
   public async querySpeakers(): Promise<AudioDeviceInfo[]> {
-    return this.deviceManager.getSpeakers();
+    try {
+      return this.deviceManager.getSpeakers();
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
+      return Promise.resolve([] as AudioDeviceInfo[]);
+    }
   }
 
   public async askDevicePermission(constrain: PermissionConstraints): Promise<void> {
-    await this.deviceManager.askDevicePermission(constrain);
+    try {
+      await this.deviceManager.askDevicePermission(constrain);
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
+    }
   }
 
   public async joinCall(microphoneOn?: boolean): Promise<void> {
     if (isInCall(this.getState().call?.state ?? 'None')) {
-      throw new Error('You are already in the call!');
+      const e = new Error('You are already in the call!');
```

Comment:
nit: this error should indicate this is a joining error, `Error joining call - user is already in the call`.

Reply:
Sounds good, updated to be `Error joining call - user is already in the call`